plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2733,6 +2747,30 @@ private static Optional<String> find(Map<String, ColumnHandle> assignments, Proj
         return Optional.empty();
     }
 
+    @Override
+    public void validateScan(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        DeltaLakeTableHandle deltaLakeTableHandle = (DeltaLakeTableHandle) handle;
+
+        if (isQueryPartitionFilterRequired(session)) {
+            List<String> partitionColumns = deltaLakeTableHandle.getMetadataEntry().getCanonicalPartitionColumns();
+            if (!partitionColumns.isEmpty()) {
+                if (deltaLakeTableHandle.getAnalyzeHandle().isPresent()) {
+                    throw new TrinoException(QUERY_REJECTED, "The session with enabled property delta.query-partition-filter-required does not work with ANALYZE statement.");
```

Comment:
Can we re-word the error as `Session with delta.query-partition-filter-required enabled does not work with ANALYZE statement`

WDYT @findinpath

Reply:
`delta.query-partition-filter-required` is not a session option, but rather a configuration option.

cc @marcinsbd

Reply:
nevertheless, @Praveen2112 , @findinpath ,  is the exception message OK? if not pls provide how it's supposed be formulated

Reply:
My take on the exception message:

```suggestion
                    throw new TrinoException(QUERY_REJECTED, "ANALYZE statement can't be performed on partitioned tables because filtering is required on at least one partition. The partition filtering check can be however disabled with the catalog session property 'query_partition_filter_required'.");
```